### ISSUE TITLE
touch support in wayland (gnome, phosh)

### ISSUE THREAD

**Issue by subjectdenied** [Genymobile/scrcpy#1536]
- [x] I have read the [FAQ](https://github.com/Genymobile/scrcpy/blob/master/FAQ.md).
 - [x] I have searched in existing [issues](https://github.com/Genymobile/scrcpy/issues).

**Environment**
 - OS: Mobian (Debian Linux for Pinephone)
 - scrcpy version: latest
 - installation method: apt
 - device model: -
 - Android version: -

**Describe the bug**
using phosh (phone sized wayland client) on my pinephone no touch events are registered in scrcpy, while it is working flawlessly with X11. i tested this also under gnome using wayland, and touch is also not recognized

**Comment by rom1v** [comment]
Oh, that's unfortunate. :disappointed: 

However, scrcpy uses [SDL](https://libsdl.org/) to receive input events (`SDL_TouchFingerEvent`), so the problem should be reported against libsdl.

**Comment by subjectdenied** [comment]
i asked in #SDL on IRC and got a reply that touch might be only implemented for wayland when using QT

**Comment by rom1v** [comment]
> on further investigation i found that on wayland touch is actually working, however the surface that recognizes touch seems to be scaled down to a small area of the screen (i'd say a 1/8) compared to the real window-dimensions

Oh, interesting! And the mouse works as expected?

It might be #15, but specific to touch events.

Is it better with this change?

```diff
diff --git a/app/src/input_manager.c b/app/src/input_manager.c
index e8ba9f79..588432c6 100644
--- a/app/src/input_manager.c
+++ b/app/src/input_manager.c
@@ -467,7 +467,7 @@ convert_touch(const SDL_TouchFingerEvent *from, struct screen *screen,
 
     int ww;
     int wh;
-    SDL_GL_GetDrawableSize(screen->window, &ww, &wh);
+    SDL_GetWindowSize(screen->window, &ww, &wh);
 
     // SDL touch event coordinates are normalized in the range [0; 1]
     int32_t x = from->x * ww;
```

**Comment by subjectdenied** [comment]
this actually fixed it, thanks for looking into it! 

will test mouse later also

**Comment by rom1v** [comment]
OK, thank you, so I'll write a better fix. I'll ask you to test again when it's done :)

**Comment by rom1v** [comment]
Here is a proper fix: fb5e619636299af06e8f71df8a71fe620cadca7f

Could you confirm that is works, please?

**Comment by subjectdenied** [comment]
working perfect, thanks for looking into this!

**Comment by rom1v** [comment]
Thank you for your report and feedback :+1:

Merged into `dev`: e8a565f9ea91a30cd354de696b3b35ea74cda83b
